metrics/metrics.go

```diff
@@ -22,6 +22,8 @@ type Recorder interface {
 	IncBulkheadTimeout()
 	// IncCircuitbreakerState increments the number of state change.
 	IncCircuitbreakerState(state string)
+	// SetCircuitbreakerCurrentCondition sets the condition of ciruit breaker.
+	SetCircuitbreakerCurrentCondition(condition int)
```

Comment:
Would be better to have uniformity between method signatures:
- On name: `SetCircuitbreakerCurrentCondition` -> `SetCircuitbreakerCurrentState`
- On parameter: `condition int` -> `state string`

Reply:
Will make the name change. 
I suspect the parameter change will fall out of the refactoring the metric approach. Stay tuned!